services/app-api/src/postgres/contractAndRates/prismaDraftRatesHelpers.ts

```diff
@@ -0,0 +1,54 @@
+import { Prisma } from '@prisma/client'
+import {
+    ContractRevisionType,
+    RateRevisionWithContractsType,
+} from '../../domain-models/contractAndRates'
+import { contractRevisionToDomainModel } from './parseContractWithHistory'
+import {
+    includeContractFormData,
+    includeUpdateInfo,
+    rateFormDataToDomainModel,
+} from './prismaSharedContractRateHelpers'
+import { RateRevisionTableWithContracts } from './prismaSubmittedRateHelpers'
+
+const includeDraftContracts = {
+    revisions: {
+        include: {
+            ...includeContractFormData,
+            submitInfo: includeUpdateInfo,
+            unlockInfo: includeUpdateInfo,
+        },
+        take: 1,
+        orderBy: {
+            createdAt: 'desc',
+        },
+    },
+} satisfies Prisma.ContractTableInclude
+
+type DraftContractsTable = Prisma.ContractTableGetPayload<{
+    include: typeof includeDraftContracts
+}>
+
+function draftContractsToDomainModel(
+    draftContracts: DraftContractsTable[]
+): ContractRevisionType[] {
+    return draftContracts.map((dc) =>
+        contractRevisionToDomainModel(dc.revisions[0])
```

Comment:
Nice this is good - I like this way of making sure we only grab the first by putting it just at the level of the domain transform function